crates/cli-support/src/js/mod.rs

```diff
@@ -310,11 +316,31 @@ impl<'a> Context<'a> {
             js = js.replace("\n\n\n", "\n\n");
         }
 
-        Ok((js, ts))
+        let preloads = if self.config.mode.electron_renderer() {
+            preloads.push_str("\nprocess.once('loaded', () => {\n");
+            for (_module, items) in crate::sorted_iter(&self.js_imports) {
+                for (_i, (item, rename)) in items.iter().enumerate() {
+                    let mut name: &str = item;
+                    if let Some(other) = rename {
+                        name = other;
+                    }
+                    preloads.push_str("  global.");
+                    preloads.push_str(name);
+                    preloads.push_str(" = ");
+                    preloads.push_str(name);
+                    preloads.push_str(";\n");
+                }
+            }
+            preloads.push_str("});");
+            Some(preloads)
+        } else {
+            None
+        };
+
+        Ok((js, ts, preloads))
     }
 
-    fn js_import_header(&self) -> Result<String, Error> {
-        let mut imports = String::new();
+    fn js_import_header(&self, imports: &mut String, preloads: &mut String) -> Result<(), Error> {
```

Comment:
Instead of taking two out-parameters could this return a structure with named fields that document the meaning of each field?